docs: Expand the website readme (#3412)
weaveworks/weave-gitops: website/README.md

> # WeaveGitOps Product Website

Reviewer: ```suggestion
# Weave GitOps Product Website
```

This website is built using [Docusaurus 2](https://docusaurus.io/), a modern static website generator.
It requires Node.js installed on your system. Install or Update your current setup for Node.js from [here](https://nodejs.org/en/download/)
Install yarn from [here](https://classic.yarnpkg.com/en/docs/install)

> The docs are live [here](https://docs.gitops.weave.works/).

Reviewer: ```suggestion
The website is live [here](https://docs.gitops.weave.works/).
```

## Developing the docs

### Start a local development server

> 1. Change into the `website` directory `cd website`.

Reviewer: ```suggestion
1. Change into the `website` directory: `cd website`.
```

> 1. Install dependencies `yarn install`.

Reviewer: ```suggestion
1. Install dependencies: `yarn install`.
```

> 1. Set a fake Google Analytics API key to pass validation errors.

Reviewer: ```suggestion
1. Set a fake Google Analytics API key to pass validation errors: `export GA_KEY=fakekey`.
```

>     ```bash
>     export GA_KEY=fakekey
>     ```
> 

Reviewer: ```suggestion
```

that change will be present for all versions.

#### How to add yaml/json/console output without making it painful

Our product comes with a lot of `yaml`, and we need to document that. However,
too much can saturate a page and make it hard to read. It can also be easy to
miss bits of text lying in between large chunks of `yaml`. The solution is to
collapse large `yaml`, `json` or console output sections with `<details>` tags.

Smaller chunks are usually fine. But for anything more that 10 lines it is
neater to wrap it.

~~~markdown
<details><summary>Expand to see this awesome HelmRepository</summary>

Don't forget to leave a line break below the opening tag...
```yaml
---
woohoo:
  such: yaml
```
and before the closing one.

</details>
~~~

<details><summary>Expand to see this awesome HelmRepository</summary>

Don't forget to leave a line break below the opening tag...
```
---
woohoo:
  such: yaml
```
and before the closing one.

</details>

#### How to add screenshots

Please, please, zoom in before you take the shot. I promise they don't need to
see the whole page, just the bit you are talking about is fine.

### After you have finished your changes

> Check that the docs build with:
> 
> ```bash
> export GA_KEY=fakekey

Reviewer: ```suggestion
Check that the docs build with `GA_KEY=fakekey yarn build`
```

yarn build
```

The build will also be run on the PR, but it is convenient to run it before you
open one.

## Deploying the docs

Do not attempt to deploy the docs manually.

Deployment happens automatically (once the tests pass) upon merging to the

> default branch: see [.github/workflows/docs.yaml](.github/workflows/docs.yaml) for config.

Reviewer: ```suggestion
`main` branch: see [.github/workflows/docs.yaml](.github/workflows/docs.yaml) for config.
```
more explicit so no one has to make up their minds about what the default branch is.
